docs/design_docs/shared_steps.md

```diff
@@ -0,0 +1,70 @@
+# Shared steps
+
+date: 2023/08/18
+
+Contributors: Carolyn Begeman, Xylar Asay-Davis
+
+## Summary
+
+The capability designed here is the ability to share steps across test cases.
+In this design document, "shared steps" refers to any step which may be used by
+multiple test cases that are available in polaris.
+
+The main motivation behind this capability is the computational expense of
+running steps that could shared across test cases multiple times. In order to
+reflect the fact that steps are shared to the user, we present a new design for
+the working directory structure. The design is successful insofar as it
+guarantees that shared steps are run once per slurm job and that the role of
+shared steps is clear to users.
+
+## Requirements
+
+### Requirement: Shared steps are set up once (optional).
+
+Ideally, steps used by multiple test cases should be run once per invocation of
+`polaris setup` or `polaris suite`. This requirement may be dropped since step
+set up is not generally resource intensive.
+
+### Requirement: Shared steps are run once.
+
+Shared steps should be per run once per invocation of `polaris serial` or
+`polaris parallel`.
+
+### Requirement: Shared steps are run before steps that depend on their output.
+
+### Requirement: Object properties of shared steps are intuitive.
+
+Shared steps are not daughters of test cases.
+
+### Requirement: Working directory structure is intuitive.
+
+The working directory structure should not have shared step directories as
+subdirectories of test cases.
```

Comment:
```suggestion
Shared step directories should be located at the highest level in the working directory structure where all test cases that use that step are run at or below that level.
```
@xylar I see what you mean. Is this compatible with what you were suggesting?

Reply:
I think that works well.